### PR TITLE
Add Order details to CheckoutCompleted event

### PR DESCRIPTION
**Pull request by josemiguel-alvarez**
### What are you trying to accomplish?

Add Order details to CheckoutCompleted event

### How to test

Complete a checkout and check the app logs.

#### Testing

- [x] I've added tests to support my implementation

#### Contribution guidelines

- [ ] I have read and agree with the [Contribution Guidelines](https://github.com/shopify/checkout-sheet-kit-swift/blob/main/.github/CONTRIBUTING.md).
- [ ] I have read and agree with the [Code of Conduct](https://github.com/shopify/checkout-sheet-kit-swift/blob/main/.github/CODE_OF_CONDUCT.md).

#### Documentation

- [ ] I've updated any documentation related to these changes.
- [ ] I've updated the [README](https://github.com/shopify/checkout-sheet-kit-swift).

#### Versioning

- [ ] I have bumped the version number in the [`.podspec` file](https://github.com/Shopify/checkout-kit-swift/blob/main/ShopifyCheckoutKit.podspec#L2).
- [ ] I have bumped the version number in the [README](https://github.com/Shopify/checkout-kit-swift/blob/main/README.md#packageswift).
- [ ] I have added a [Changelog](./CHANGELOG.md) entry.

### After merging

If your changes required a version bump, please add an entry under the [Releases](https://github.com/Shopify/checkout-kit-swift/releases) page - doing so will trigger a CI workflow to publish the latest Cocoapod.

### CODE REVIEW

File: Sources/ShopifyCheckoutSheetKit/CheckoutCompletedEvent.swift [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+			case cart
+			case paymentMethods
+			case billingAddress
+			case deliveries
+		}
+	}
+
+	public struct CartInfo: Decodable {
+		public let lines: [CartLine]
+		public let price: PriceSet
```
[the remark this answers is not in the document]
Comment (josemiguel-alvarez) [reply Sources/ShopifyCheckoutSheetKit/CheckoutCompletedEvent.swift:63]: It depends, do we want to make every field optional? In that case yes. If we want to match the types on `checkout-web` then no.

I don't think it makes sense to mark everything as optional as the SDK schema on `checkout-web` is versioned, so it shouldn't change.

File: Sources/ShopifyCheckoutSheetKit/CheckoutCompletedEventDecoder.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+import Foundation
+
+class CheckoutCompletedEventDecoder {
+	func decode(from container: KeyedDecodingContainer<CheckoutBridge.WebEvent.CodingKeys>, using decoder: Decoder) throws -> CheckoutCompletedEvent {
```
[the remark this answers is not in the document]
Comment (josemiguel-alvarez) [reply Sources/ShopifyCheckoutSheetKit/CheckoutCompletedEventDecoder.swift:27]: It throws. `try` just declares that an error can be thrown in this line. This documentation explains it: https://docs.swift.org/swift-book/documentation/the-swift-programming-language/errorhandling/#Handling-Errors

File: Tests/ShopifyCheckoutSheetKitTests/CheckoutBridgeTests.swift [hunk 3]
```diff
@@ -70,18 +70,30 @@ class CheckoutBridgeTests: XCTestCase {
 		}
 	}
 
-	func testDecodeSupportsCheckoutCompleteEvent() throws {
+	func testDecodeSupportsCheckoutCompletedEvent() throws {
```
[the remark this answers is not in the document]
Comment (josemiguel-alvarez) [reply Tests/ShopifyCheckoutSheetKitTests/CheckoutBridgeTests.swift:73]: We already have some test checking the correct behavior of a failure when decoding an event. It's not necessary to add a new one as it will be the same and the functionality that handles failures is related to CheckoutBridge, not the CheckoutCompleted event.

As every field is optional now there's also no needed of adding more tests for partial payload or empty payload.

File: Sources/ShopifyCheckoutSheetKit/CheckoutDelegate.swift [hunk 4]
```diff
@@ -27,7 +27,7 @@ import UIKit
 /// A delegate protocol for managing checkout lifecycle events.
 public protocol CheckoutDelegate: AnyObject {
 	/// Tells the delegate that the checkout successfully completed.
-	func checkoutDidComplete()
+	func checkoutDidComplete(event: CheckoutCompletedEvent)
```
Comment (markmur) [inline Sources/ShopifyCheckoutSheetKit/CheckoutDelegate.swift:30]: Since we're updating the protocol here, this will be a breaking change and will require a major version bump. 

We could avoid a major bump by deprecating this method and creating a new one with the event arg:

```swift
@deprecated
func  checkoutDidComplete()

func checkoutDidCompleteWithEvent(event: CheckoutCompletedEvent)
```

@kiftio we don't have the option to overload here in Swift. Is that what you did in the Android repo?
